src/libraries/System.Memory.Data/tests/BinaryDataTests.cs

```diff
@@ -431,6 +431,73 @@ public void MaxStreamLengthRespected()
             var data = BinaryData.FromStream(new OverFlowStream(offset: int.MaxValue - 1000));
         }
 
+        [Fact]
+        public async Task CanCreateBinaryDataFromFile()
+        {
+            byte[] buffer = "some data"u8.ToArray();
+            string path = Path.GetTempFileName();
+            try
+            {
+                File.WriteAllBytes(path, buffer);
+                BinaryData data = BinaryData.FromFile(path);
+                Assert.Equal(buffer, data.ToArray());
+
+                byte[] output = new byte[buffer.Length];
+                var outputStream = data.ToStream();
+                outputStream.Read(output, 0, (int)outputStream.Length);
+                Assert.Equal(buffer, output);
+
+                data = await BinaryData.FromFileAsync(path);
+                Assert.Equal(buffer, data.ToArray());
+
+                outputStream = data.ToStream();
+                outputStream.Read(output, 0, (int)outputStream.Length);
+                Assert.Equal(buffer, output);
+            }
+            finally
+            {
+                File.Delete(path);
+            }
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData(MediaTypeNames.Application.Soap)]
+        public async Task CanCreateBinaryDataFromFileWithMediaType(string? mediaType)
+        {
+            byte[] buffer = "some data"u8.ToArray();
+            string path = Path.GetTempFileName();
+            try
+            {
+                File.WriteAllBytes(path, buffer);
+                BinaryData data = BinaryData.FromFile(path, mediaType);
+                Assert.Equal(buffer, data.ToArray());
+                Assert.Equal(mediaType, data.MediaType);
+
+                byte[] output = new byte[buffer.Length];
+                var outputStream = data.ToStream();
+                outputStream.Read(output, 0, (int)outputStream.Length);
+                Assert.Equal(buffer, output);
+
+                data = await BinaryData.FromFileAsync(path, mediaType);
+                Assert.Equal(buffer, data.ToArray());
+                Assert.Equal(mediaType, data.MediaType);
+
+                outputStream = data.ToStream();
+                outputStream.Read(output, 0, (int)outputStream.Length);
+                Assert.Equal(buffer, output);
+
+                //changing the backing buffer should not affect the BD instance
+                buffer[3] = (byte)'z';
+                Assert.NotEqual(buffer, data.ToMemory().ToArray());
+            }
+            finally
+            {
+                File.Delete(path);
```

Comment:
Can we use the FileCleanupTestBase base class for these tests instead of doing the manual cleanup?

Reply:
Yes, I have moved these tests to a dedicated test class to avoid creating and deleting a test directory in all others tests for nothing.